### DISCUSSION TITLE
Get child PIXI.Graphics by coords

### DISCUSSION THREAD

**Discussion by Nawarius** [pixijs/pixijs#7970] [Q&A] [answered]
Hello!

I organize the collision system using PIXI.Graphics invisible ellipses that copy the position of the sprites on the background. 
`new PIXI.Graphics.drawEllipse(x, y, w, h)`

Everything works very well, my character finds these ellipses correctly and collides with them. The only thing I'm a little worried about is optimization, so this is a question about it.

The fact is that I am looking for these ellipses, going through a loop like this on each iteration of the pixi ticker.
```
for (let ellipse of background.children) {
   if (ellipse.containsPoint({x, y})) collisionExist = true
}
```

I don't like to iterate all children on every pixi ticker tick. What could save the day is some kind of function that would help me find the ellipse by coordinate.

`containsPoint` works perfect. But it only works for an ellipse. I would like to be able to check whether some kind of graphics exists for specific coordinates or does not exist

**Comment by eXponenta** [comment, marked as answer]
Anyway containsPoint will be executed.

There are not another way to find anything without iterations in naive implementation.
Interaction doing same for checking collision between cursor and element - it recursive execute containsPoint.

For more faster iterations you should use spatial maps, like quad tree or greeded hash map.
What you want to use - your problem, because this is dependent of how many elements you should check, how many elements is movable etc.

For expensive physics you should drop naive implementation and use already optimised libs like Box2D, nape, Cannon
